app/scripts/controllers/pane-articles-lists.js

```diff
@@ -0,0 +1,192 @@
+    'use strict';
+
+    /**
+    * Constructor function for the articles lists modal controller.
+    *
+    * @class ModalCtrl
+    */
+    function ModalCtrl($modalInstance, $sce, info, $rootScope) {
+        var self = this,
+            url;
+
+        switch(info.action) {
+            case 'edit':
+                url = [
+                    AES_SETTINGS.API.rootURI,
+                    '/admin/playlists/',
+                    info.articleId,
+                    '/',
+                    info.language,
+                    '/editor-view/', info.articlesListId
+                ].join('');
+
+                break;
+            case 'attach':
+                url = [
+                    AES_SETTINGS.API.rootURI,
+                    '/admin/playlists/',
+                    info.articleId,
+                    '/',
+                    info.language,
+                    '/editor-view'
+                ].join('');
+
+                break;
+            default:
+                url = '';
+        }
+
+        self.url = $sce.trustAsResourceUrl(url);
+
+        /**
+        * Closes the modal.
+        *
+        * @method close
+        */
+        self.close = function () {
+            $rootScope.$broadcast('close-articles-lists-modal');
+            $modalInstance.close();
+        };
+    }
+
+    /**
+    * AngularJS controller for the ArticlesLists pane.
+    *
+    * @class PaneArticlesListsCtrl
+    */
+    angular.module('authoringEnvironmentApp')
+        .controller('PaneArticlesListsCtrl', [
+        '$q',
+        '$modal',
+        '$scope',
+        'article',
+        'modalFactory',
+        'ArticlesList',
+        'toaster',
+        'TranslationService',
+        function (
+            $q,
+            $modal,
+            $scope,
+            articleService,
+            modalFactory,
+            ArticlesList,
+            toaster,
+            TranslationService) {
+            var self = this,
+                article = articleService.articleInstance;
+
+            // retrieve all articlesLists assigned to the article
+            self.assignedArticlesLists = ArticlesList.getAllByArticle(
+                article.articleId, article.language
+            );
+
+            $scope.$on('close-articles-lists-modal', function(event) {
+                ArticlesList.getAllByArticle(
+                    article.articleId, article.language
+                ).$promise.then(function (data) {
+                    // removes list from assignedArticlesLists
+                    // if the list doesn't exist in the response data
+                    angular.forEach(
+                        self.assignedArticlesLists,
+                        function (value, key) {
+                            if (!_.some(data, {id: value.id})) {
+                                _.remove(
+                                    self.assignedArticlesLists,
+                                    {id: value.id}
+                                );
```

Comment:
good point, although it worked fine, but it was not a good approach, I replaced that snippet with the iteration in reverse + I have added more tests to cover this